### PR TITLE
Variable tab widget

### PR DESCRIPTION
**Pull request by JohannesLorenz**
This adds a boolean to the tab widget to specify whether it should be resizable. As a consequence, the Instrument Window will resize to fit its contents.

### CODE REVIEW

**Comment by JohannesLorenz** [comment]
This may be the most difficult Lv2 sidebranch concerning regression (i.e. other ). It would be really cool if someone can review, otherwise I'll need to merge this without review in 7 days.

File: src/tracks/InstrumentTrack.cpp [hunk 1]
```diff
@@ -1849,4 +1855,9 @@ void InstrumentTrackWindow::viewPrevInstrument()
 	viewInstrumentInDirection(-1);
 }
 
+void InstrumentTrackWindow::adjustTabSize(QWidget *w)
+{
+	w->setMinimumSize(INSTRUMENT_WIDTH - 4, INSTRUMENT_HEIGHT - 4);
```
[the remark this answers is not in the document]
Comment (JohannesLorenz) [reply src/tracks/InstrumentTrack.cpp]: If you look very close at the Instrument's plugin tab, you see that the instrument itself (orange in the image) has a left and right offset of 2 (i.e. 2+2=4). This is done by moving the widget [in TabWidget.cpp, l. 89](https://github.com/JohannesLorenz/lmms/blob/d52c220a5cd40af37a0c167c1425f89edca991cf/src/gui/widgets/TabWidget.cpp#L89). This also explains why LMMS uses `INSTRUMENT_WIDTH	= 254`, but the artworks are all 250x250.

**Comment by JohannesLorenz** [comment]
First of all, many thanks for the review!

I fixed everything, any stripes (like the orange one for 3osc) should be gone now.

I'll do a regular merge when you're OK.

**Comment by CYBERDEViLNL** [comment]
> First of all, many thanks for the review!
> 
> I fixed everything, any stripes (like the orange one for 3osc) should be gone now.
> 
> I'll do a regular merge when you're OK.

Awesome! Yes I'm happy with the code :+1:

**Comment by PhysSong** [comment]
From #4899:
>     * Known Bug: Instrument windows are resized wrong in some cases when using the arrows to switch instruments in the instrument window. This bug does only affect projects with at least one "variable sized instrument".

Isn't this related to this PR? It means this PR doesn't break anything for now since we don't use this feature right now, but there are some issues with the code.

**Comment by JohannesLorenz** [comment]
@PhysSong Unfortunately that's true. The code will not break anything, but will have this UI issue. I tried to fix it, but I really did not understand it. For now, I think Lv2 is more urgent than fixing that minor UI bug.

**Pull request merged**